### ISSUE TITLE
Analyses and reductions in MEI

### ISSUE THREAD

**Issue by pettter** [music-encoding/music-encoding#786]
I am working on a way to annotate MEI scores with graph (including tree) based analyses, in particular including Schenker-like successive reductions. For analyses involving the visible notes I've begun using a variant of the standard proposed by Rizo and Marsden in [this article](https://link.springer.com/article/10.1007/s00799-018-0262-x)(is that on track to be merged into the standard, by the way?), but for a true representation of the relationships one might want to represent, it may be necessary to add new notes that are not present in the original score. I would like to keep the analysis (including any added notes) separate from the original score, if possible, such that the analysed score can be retrieved and rendered without the analysis interfering.

So, the question for me is how to encode that properly. One relatively simple option is to stick new `note`s in the respective `label` elements of the `node`s in the `graph` directly, but instead of using `sameas` to whatever surface-level `note` the node represents, I create a new `note` that is present only in the `graph`, with either `follows`, `precedes` or `synch` to ground them in the score, for rendering the reduction(s).

Another option is to create entirely new `score` elements representing each reduction (with the appropriate metadata in the `scoreDef`), containing only the notes present in that layer, including any new ones (linking back to the original `score` using `corresp` or `copyof` depending on if it's a literal copy or with some modification), which would help with rendering those reductions properly, since it would allow for a lot more control overall.

Any input (including words to the effect of "this is out of scope for MEI") would be greatly appreciated.

**Comment by kepper** [comment]
Hi Petter, I'm not perfectly comfortable with an answer here, but maybe we can work this out together. First of all, the structures in the article you've mentioned haven't been included in MEI yet. At least to my knowledge, there was never a real dicussion _within in the community_. I would assume that this is something the Analysis IG should discuss and then a corresponding customization for MEI can be fleshed out (I'd be happy to contribute to that…). Accordingly, most of the elements mentioned cannot be used (yet?) in regular MEI, so this discussion is somewhat arcane until we have a concrete model in hand. 

There are some basic things than can be addressed already, though. `@sameas` is a bit tricky. The documentation for it reads that it "points to an element that is the same as the current element but is not a literal copy of the current element." Usually, it is used to deal with elements that are visible only once, but for some reason need to be encoded twice. An example would be a rest that is shared between two layers on the same staff – when you don't want to duplicate `@dur` etc. on both, you just encode that on one of the elements, and put a `@sameas` on the second. 
`@copyof` is actually sometimes used as an encoding shortcut: When resolving a copyof from one measure to the other, I would copy the markup tree and adjust all `@xml:id`s. I'm aware that `@copyof` can be used in a less technical, more semantic / analytical way, and that this should be clarified in the documentation. I'm not saying that your use of `@copyof` is incorrect, I'm just saying that it could be interpreted differently than you expect. Finally, `@corresp` is an unspecified relation between any two elements. Accordingly, it may be used for your purpose, and there is absolutely nothing wrong in doing just that. However, it's an unspecified connection, and as such may be interpreted differently than you expect. A more precise encoding can be achieved using the `<annot>` element with `@plist` to identify one end, a child `<relation>` with `@rel` indicating the type of relationship, and `@target` to identify the other end of your relation. The concept is pretty much RDF-like, and creates significant amounts of markup, but only with that you can ensure that everyone will understand your markup as it's meant. 

Especially when using `<annot>` for linking, using separate `<score>`s seems like a plausible idea (though you probably want to wrap that in repeated `<mdiv>`s or similar instead – you could actually have separate files for each reduction – this may require more considerations…). It depends on whether you plan to render the reductions on top of the full score, or if they're planned to be used independent of that only. 

Not really a solution yet, but maybe something to get the discussion started? And no worries, it's definitely relevant 👍

**Comment by pettter** [comment]
Glad to hear it!

Gotcha. I am, to my shame, not validating the files I'm generating yet, and Verovio is kind enough to just give a warning on render (though stripping it out would be Easy(tm), but I'll keep it in mind to push for a proposal in the IG. To me, it would make sense to more or less directly lift the TEI definitions, since it's such a generic structure, but possibly there should be some standard ways of referring to the analysed things, whether they be individual notes, measures, harmonic symbols, timespans, or what-have-you. At the very least, there needs to be documentation and the like.

That's very useful info about the nuances of `@corresp`, `@copyof` and `@sameas`. I've used `@sameas` in the nodes of my graph analyses for now, since that's what Rizo and Marsden used in their examples, but for the score reductions I noticed some of the engraving issues already. Thanks!

To me, using a separate `<mdiv>` seems counterintuitive - an analysis or reduction is not a separate piece, but just a different view of the _same_ piece - indeed the same "score", even. This might be a more philosophical issue however, where I am more likely to look at the abstract, theoretical piece of which every score is a representation, and the `<mdiv>` and MEI view is more towards looking at concrete scores and digital representations of the _score_ rather than the _piece_. Using a separate `<mdiv>` would also reduce or (maybe) remove the need to modify the MEI standard, which I can only see as a good thing.

(As an aside: separating the analysis/reduction to a separate file is something I'm doing my best to avoid)

With using a separate `<mdiv>`, do you think it would be enough to indicate in the `<scoreDef>` that "this is a reduction of that other `<score>`, and we're using `@corresp` to indicate elements which are linked" ? The RDF-like syntax seems, as you say, to use significant amounts of markup for, to me at least, not much semantic gain. Or do you think the problem would be that there are situations that would call for the use of `@corresp` links also within the same (reduced) `<score>`?

**Comment by ahankinson** [comment]
I'm going to move this to the Discussions section.
